ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -285,6 +285,15 @@ static void setPurgeUptoSnapshotIndex(RaftProperties properties, boolean shouldP
       setBoolean(properties::setBoolean, PURGE_UPTO_SNAPSHOT_INDEX_KEY, shouldPurgeUptoSnapshotIndex);
     }
 
+    String PURGE_PRESERVE_LOG_NUM_KEY = PREFIX + "purge.preserve.log.num";
```

Comment:
"preserve" is a good word choice.  Let's call it ".purge. preservation" so that it is similar to ".purge.gap".

BTW, it needs a "." in the beginning.

Reply:
Done